src/pipeline.cc

```diff
@@ -149,6 +155,84 @@ Result Pipeline::SetShaderCompileOptions(const Shader* shader,
                 shader->GetName());
 }
 
+Result Pipeline::SetShaderRequiredSubgroupSize(const Shader* shader,
+                                               const ShaderInfo::RequiredSubgroupSizeSetting subgroupSizeSetting) {
+  if (!shader)
+    return Result("invalid shader specified for  required subgroup size");
+
+  for (auto& info : shaders_) {
+    const auto* is = info.GetShader();
+    if (is == shader) {
+      info.SetRequiredSubgroupSizeSetting(subgroupSizeSetting);
+      return {};
+    }
+  }
+
+  return Result("unknown shader specified for required subgroup size: " +
+                shader->GetName());
+}
+
+Result Pipeline::SetShaderRequiredSubgroupSize(const Shader* shader,
+    const uint32_t subgroupSize)
+{
+    const bool isPow2 = (subgroupSize & (subgroupSize - 1)) == 0;
+    if (subgroupSize == 0 || subgroupSize > 128 || !isPow2) {
+        return Result("invalid required subgroup size " +
+            std::to_string(subgroupSize) +
+            " specified for shader name " +
+            shader->GetName());
+    }
+    const ShaderInfo::RequiredSubgroupSizeSetting subgroupSizeSetting =
+        static_cast<ShaderInfo::RequiredSubgroupSizeSetting>(subgroupSize);
```

Comment:
I'd prefer this to be done a different way. Maybe we store a flag if it's numeric, max or min and then a second value which stores the numeric.

Reply:
Changed to an enum + numeric value.